crypto/evp/exchange.c

```diff
@@ -229,9 +229,12 @@ int EVP_PKEY_derive_init(EVP_PKEY_CTX *ctx)
 
     /*
      * Because we cleared out old ops, we shouldn't need to worry about
-     * checking if exchange is already there.
+     * checking if exchange is already there. The fetch can fail in the legacy
+     * case.
      */
+    ERR_set_mark();
     exchange = EVP_KEYEXCH_fetch(ctx->libctx, supported_exch, ctx->propquery);
+    ERR_pop_to_mark();
```

Comment:
If you look, you'll find `ERR_set_mark()` and `ERR_pop_to_mark()` are already used, but not so tightly around this specific call.

Reply:
That goes for all places you added them

Reply:
Ah - thanks. I missed that. I rolled this change back - and a number of the others.

Reply:
It seems you forgot about this